types/index.ts

```diff
@@ -54,82 +54,70 @@ export interface MatchedAudience {
   matchedOnCurrentPageView: boolean;
 }
 
+export type AudienceState = 'live' | 'paused' | 'deleted';
+
+export interface CachedAudienceMetaData {
+  cachedAt: number;
+  audiences: AudienceMetaData[];
+}
+
+export interface AudienceMetaData {
+  id: string;
+  version: number;
+}
+
 export type StringArrayQueryValue = string[];
+
 export type VectorQueryValue = {
   vector: number[];
   threshold: number;
 };
 
-export type AudienceState = 'live' | 'paused' | 'deleted';
+export enum QueryFilterComparisonType {
+  VECTOR_DISTANCE = 'vectorDistance',
+  COSINE_SIMILARITY = 'cosineSimilarity',
+  ARRAY_INTERSECTS = 'arrayIntersects',
+}
 
-export interface AudienceDefinition {
-  accountId?: Record<string, AudienceState>;
-  id: string;
-  version: number;
-  name: string;
-  cacheFor?: number;
-  definition:
-    | {
-        featureVersion: number;
-        ttl: number;
-        lookBack: number;
-        occurrences: number;
-        queryProperty: string;
-        queryValue: VectorQueryValue;
-        queryFilterComparisonType: 'vectorDistance';
-      }
-    | {
-        featureVersion: number;
-        ttl: number;
-        lookBack: number;
-        occurrences: number;
-        queryProperty: string;
-        queryValue: VectorQueryValue;
-        queryFilterComparisonType: 'cosineSimilarity';
-      }
-    | {
-        featureVersion: number;
-        ttl: number;
-        lookBack: number;
-        occurrences: number;
-        queryProperty: string;
-        queryValue: StringArrayQueryValue;
-        queryFilterComparisonType: 'arrayIntersects';
-      };
+export interface ArrayIntersectsFilter {
+  queryValue: StringArrayQueryValue;
+  queryFilterComparisonType: QueryFilterComparisonType.ARRAY_INTERSECTS;
 }
 
-export interface CachedAudienceMetaData {
-  cachedAt: number;
-  audiences: AudienceMetaData[];
+export interface VectorDistanceFilter {
+  queryValue: VectorQueryValue[];
+  queryFilterComparisonType: QueryFilterComparisonType.VECTOR_DISTANCE;
 }
 
-export interface AudienceMetaData {
+export interface CosineSimilarityFilter {
+  queryValue: VectorQueryValue[];
+  queryFilterComparisonType: QueryFilterComparisonType.COSINE_SIMILARITY;
+}
+
+export type AudienceDefinitionFilter =
+  | VectorDistanceFilter
+  | CosineSimilarityFilter
+  | ArrayIntersectsFilter
+
+export type AudienceDefinition = {
+  featureVersion: number;
+  ttl: number;
+  lookBack: number;
+  occurrences: number;
+  queryProperty: string;
+} & AudienceDefinitionFilter;
+
+export interface Audience {
```

Comment:
`AudienceDefinition` && `AudienceQueryDefinition` ?

I think just Audience is a little loose?

WDYT?

Reply:
Agree!